### PR TITLE
43 fullstack login

### PR DESCRIPTION
**Pull request by dylan-mankin**

### CODE REVIEW

File: backend/endpoints/.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#JWT_SECRET temporary key
```
Comment (mileswatson) [inline backend/endpoints/.env:1]: Should be in config/.env

File: backend/endpoints/auth.py [hunk 2]
```diff
@@ -2,11 +2,22 @@
...
-            return "user123"
-        else:
+        try:
+            # JWT secret key is set up in .env
+            load_dotenv()   # type: ignore
+            JWT_SIGNING_KEY: str = os.getenv(JWT_SIGNING_KEY)    # type: ignore
+            payload = jwt.decode(token, JWT_SIGNING_KEY, algorithms=[
+                                 "HS256"])
+            username: Optional[str] = payload.get("sub")
+            if username is None:
```
Comment (mileswatson) [inline backend/endpoints/auth.py:19]: I think you can just return username here?

File: backend/endpoints/api.py [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+from ninja import NinjaAPI, Schema
 from ninja.errors import AuthenticationError, ValidationError
 from django.http.request import HttpRequest
 from django.http.response import HttpResponse
+from ninja.security import HttpBearer
+
+import jwt
+import datetime
+import os
+from dotenv import load_dotenv       # type: ignore
```
Comment (mileswatson) [inline backend/endpoints/api.py:10]: Use inbuilt config/env.py instead.

File: backend/endpoints/api.py [hunk 4]
```diff
@@ -30,3 +36,40 @@ def validation_error(request: HttpRequest, e: ValidationError) -> HttpResponse:
...
+    # check password
+    token = create_token(username)
+    return 200, TokenSchema(access_token=token)
+
+
+def create_token(username: str) -> str:
+    load_dotenv()
+    JWT_SIGNING_KEY = str(os.getenv('JWT_SIGNING_KEY'))          # type: ignore
+    JWT_ACCESS_EXPIRY = os.getenv('JWT_ACCESS_EXPIRY')      # type: ignore
+    # JWT_SIGNING_KEY = getattr(settings, "JWT_SIGNING_KEY", None)        # add attributes to settings
```
Comment (mileswatson) [inline backend/endpoints/api.py:59]: Remove unnecessary comments.

**Review by mileswatson — APPROVED** [review]

**Review by StefanPitur — APPROVED** [review]

**Pull request merged**